test/integration/stats_integration_test.cc

```diff
@@ -337,6 +338,7 @@ TEST_P(ClusterMemoryTestRunner, MemoryLargeClusterSizeWithRealSymbolTable) {
   // 2020/05/05  10908    36345       36800   router: add InternalRedirectPolicy and predicate
   // 2020/05/13  10531    36537       36800   Refactor resource manager
   // 2020/05/20  11223    36603       36800   Add primary clusters tracking to cluster manager.
+  // 2020/05/27  11289    36603       44846   Test-only change wrapping counters in test counters
```

Comment:
hmmm...maybe we can make these memory tests not use the notifying allocator. I'd like to keep these numbers prod accurate if possible.

Reply:
We can, but we'd have to plumb the boolean through several layers of code special case every waitFor function plus do a bunch of dynamic casts in the server to switch.  Is it that much worse to have a known fixed increase?

Reply:
IMO yes: the purpose of this test is to measure memory overhead and I think we should try to do it faithfully. Is it a huge amount of work to plumb some bools through the constructors?

Reply:
another alternative is to pull the stats memory tests out of the integration-test framework. I'm not sure how difficult that would be.

Reply:
Pulling it out would be more work.
I'm just going to claim if you need real stats, the counter (and eventually gague) functions don't work.  If someone needs to check them they can implement the ugly workarounds.

Reply:
Yeah I looked at that and I agree that it's not worth it to divest it from the BaseIntegrationTest.

I'm not following you about the "if you need real stats" comment; the only function of this test is to track and understand per-cluster and per-host memory overhead of stats. 

Another option which might make it easier is to have an optional virtual method in BaseIntegrationTest which you could override to ask for real stats. That would be less toil than plumbing in a bool to ctors. WDYT?

Reply:
I'm not convinced it's less toil, and given I already did the bool-toil I'd prefer to keep this as-is?

Reply:
oh sorry didn't realize you already did this with bools; I'll take a quick look. thanks.